lib/modules/datasource/docker/index.ts

```diff
@@ -812,54 +812,144 @@ export class DockerDatasource extends Datasource {
   @cache({
     namespace: 'datasource-docker-digest',
     key: (
-      { registryUrl, packageName }: GetReleasesConfig,
+      { registryUrl, packageName, currentDigest }: DigestConfig,
       newValue?: string
     ) => {
       const newTag = newValue ?? 'latest';
       const { registryHost, dockerRepository } = getRegistryRepository(
         packageName,
         registryUrl!
       );
-      return `${registryHost}:${dockerRepository}:${newTag}`;
+      const digest = currentDigest ? '@' + currentDigest : '';
+      return `${registryHost}:${dockerRepository}:${newTag}${digest}`;
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
-      let manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        newTag,
-        'head'
-      );
-      if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
-          digest =
-            (manifestResponse.headers['docker-content-digest'] as string) ||
-            null;
-        } else {
-          logger.debug(
-            { registryHost },
-            'Missing docker content digest header, pulling full manifest'
-          );
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
```

Comment:
don't use current value, it can have changed and point to wrong digest 

```suggestion
      if (currentDigest) {
```

Reply:
`currentValue` never points to a digest unambiguously, neither could you infer `currentValue` from `currentDigest`. 

The check for `if (currentValue)` is needed to assert that the entire architecture lookup is only done **iff** `currentValue` and `currentDigest` are supplied:

- `currentDigest` alone would make no sense because you can have, e.g., `FROM gcr.io/distroless/java17@sha256:ab37242e81cbc031b2600eef4440fe87055a05c14b40686df85078cc5086c98f` which doesn't point to an architecture, so the check for `currentValue` is fully on purpose.
- `currentValue` can be undefined, so the check helps to ensure manifest list retrieval will work here:
https://github.com/renovatebot/renovate/blob/1e67faf76e93fbcb0872a98e523d29c144da394c/lib/modules/datasource/docker/index.ts#L844-L859

Reply:
the current value will probably also no longer point to the original manifest list, as most docker tags are floating.

Reply:
so you need to fetch the manifest and then the config manifest to get the original architecture

https://github.com/opencontainers/image-spec/blob/main/config.md

Reply:
both response should be cached for long time, as they can't change